toolkit/scripts/makedirs.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Usage: /path/to/dirs/to/make owner_user_name
+
+# Will recursively create all directories in the path, and chown them to the specified user
+# If the directory already exists, it will be skipped and no changes will be made.
+# New directories will be created with timestamp 0.
+
+DIR="${1}"
+USER="${2}"
+
+function make_dir_recursive {
```

Comment:
Could this be replaced with `runuser -u "$SUDO_USER" -- mkdir -p "<dir>"`?

Reply:
I'll give it a shot, would certainly simplify the code a bit.

Reply:
We lose the touch -d @0, but I'm not sure that was ever a critical featuer of this. Generally if a directory is missing then we probably need to trigger a rebuild of stuff anyways. And once a directory is created we should never need to remove it unless we run a clean.

Reply:
Seems to work.

We lose the touch -d @0, but I'm not sure that was ever a critical featuer of this. Generally if a directory is missing then we probably need to trigger a rebuild of stuff anyways. And once a directory is created we should never need to remove it unless we run a clean.